compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -150,7 +150,11 @@ struct ConvertSharedMemAllocOp : public OpRewritePattern<memref::AllocOp> {
       if (auto shapeType = llvm::dyn_cast<ShapedType>(elType))
         alignement =
             shapeType.getNumElements() * shapeType.getElementTypeBitWidth() / 8;
-      else
+      else if (elType.isIndex()) {
```

Comment:
Just for my understanding, this is to get the alignment in bytes... maybe the alignment should have been in number of elements so that all the byte adjusting happens in lowering to LLVM...